assets/src/components/disconnectedModal.tsx

```diff
@@ -1,18 +1,22 @@
 import React from "react"
 import { reload } from "../models/browser"
+import { Modal } from "@restart/ui"
 
 const DisconnectedModal = () => (
-  <>
-    <div className="c-modal">
+  <Modal
```

Comment:
Suggestion: Each place where we're using restart's `Modal`, all of the props are duplicated. What if we created our own custom modal component that abstracted this?

I would recommend calling it `<SkateModal />` or something like that, not because I think that's a good name, but because [we already have a component](https://github.com/mbta/skate/blob/9ffd12145c842c46b1a4f9058a1e2e28634a5e62/assets/src/components/modal.tsx) called `<Modal />`.


(Honestly, the existing `<Modal />` should probably be called something else, because its role in the app is to _always_ be there, and to conditionally take over the app, rendering a modal on top of Skate, but coming up with better names for that is beyond the scope of this PR, IMO)

Reply:
In the modals converted so far, this is true that the props are duplicated. But not so for a few additional modals I'm converting now, though they are very close to the same. That said, I think given the slight prop variation between them, I'd prefer to keep the component invocations as-is, or at least defer a consolidation effort

Reply:
Oh darnit, this gave me an idea 🫠

Reply:
Ok I was wondering how many modals I should split out to a separate generic modal. Here's my thinking: there are three categories of modal. Notification modals, which use the basic Restart modal, basic class stylings for "c-modal", and extra styling from "_basic_notification_modal". Preset modals, which use a generic InputModal component that can be converted to the Restart modal, but separately from the other modals because it uses a special "c-input-modal" styling. And then "Other modals" to include disconnected and loading modals, which are just styled with bare bones "c-modal". I think there's some work that can be done here to consolidate the notification and "other modals" because I don't think the styling of those are very different, but that could lead to visual changes and a conversation with design. So for now, I'd kinda like to defer that refactor.

For the time being: if I make a new parent component to represent a sort of `<SkateModal />`, then `<InactiveNotificationModal ... />` would include `<BasicNotificationModal ... />`, which would include `<SkateModal .../>` which would be an extremely simple component (simple is good, but that's an extra layer of nesting than I prefer.) And `<SkateModal />` would only be used in 3 places atm. I dunno, maybe these are fine, the file navigation would just get bothersome to me?

Reply:
Part of me thinks that we might be able to avoid at least some of the indirection you mentioned with judicious use of props - `<SkateModal inactive />`, for instance - but I agree that we don't need to block this particular PR on that.